src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -207,139 +241,229 @@ private static VersionInfo GetVersionInfo()
+            Host.AssertValue(ctx);
+
+            // *** Binary format ***
+            // <prefix handled in static Create method>
+            // <base>
+            // foreach added column
+            //   ColInfoEx
+            // foreach model
+            //   whitening matrix
+            //   recovery matrix
+
+            Host.AssertNonEmpty(ColumnPairs);
+            _infos = new ColInfo[ColumnPairs.Length];
+            for (int i = 0; i < _infos.Length; i++)
+                _infos[i] = new ColInfo(ctx);
+
+            _models = new float[ColumnPairs.Length][];
+            _invModels = new float[ColumnPairs.Length][];
+            for (int i = 0; i < ColumnPairs.Length; i++)
+            {
+                _models[i] = ctx.Reader.ReadFloatArray();
+                if (_infos[i].SaveInv)
+                    _invModels[i] = ctx.Reader.ReadFloatArray();
+            }
+        }
+
+        internal WhiteningTransform(IHostEnvironment env, Arguments args, IDataView inputData)
+            : this(env, inputData, GetColumnInfos(args))
+        {
+        }
+
+        // Factory method for SignatureLoadModel
+        internal static WhiteningTransform Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            ctx.CheckAtModel(GetVersionInfo());
+            return new WhiteningTransform(env, ctx);
+        }
+
+        // Factory method for SignatureDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+            => new WhiteningTransform(env, input, GetColumnInfos(args)).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        internal static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private static (string input, string output)[] GetColumnPairs(ColInfo[] columns)
+            => columns.Select(c => (c.Input, c.Output ?? c.Input)).ToArray();
+
+        private static ColInfo[] GetColumnInfos(Arguments args)
+            => args.Column.Select(colPair => new ColInfo(colPair, args)).ToArray();
+
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
+        {
+            var inType = inputSchema.GetColumnType(srcCol);
+            var reason = TestColumn(inType);
+            if (reason != null)
+                throw Host.ExceptParam(nameof(inputSchema), reason);
+        }
+
+        internal static string TestColumn(ColumnType t)
+        {
+            if ((t.IsVector && !t.IsKnownSizeVector) || t.ItemType != NumberType.R4)
+                return "Expected float or float vector of known size";
```

Comment:
Column in TLC can be multiple-dimension tensor. However, whitening only supports 1-D vector. Could you make this check more strict? Thanks.